### PR TITLE
Log rank test

### PR DESCRIPTION
**Pull request by kshedden**
This is an implementation of log-rank testing, with tests.

### CODE REVIEW

File: src/SurvivalEstimator.jl [hunk 1]
```diff
@@ -338,6 +338,128 @@ function StatsBase.fit!(obj::KaplanMeier, Y::RCSurv)
     )
 end
 
+#-------------------
+# Log-rank tests
+#-------------------
+function update_wt(wt, nevents, nrisk, wtmethod)
+    if wtmethod == :logrank
+        return 1
```
Comment (mschauer) [inline src/SurvivalEstimator.jl:346]:
```suggestion
        return 1.0
```

File: src/Surv.jl [hunk 2]
```diff
@@ -363,6 +363,42 @@ function _tabulate_surv(T, Δ)
             noutcomes = noutcomes)
 end
 
+# Expand the time points in 'stats' to include all times in 'times'.
+function _padstats(stats, times)
```
Comment (RaphaelS1) [inline src/Surv.jl:367]: I don't understand what this does and I'm not sure it works? See comment below in tests

File: test/SurvivalEstimator.jl [hunk 3]
```diff
@@ -4,6 +4,110 @@ n = 50
...
+    Δ1 = [1, 1, 0, 1, 1]
+    T2 = Float64[2, 10, 4, 8, 6, 12]
+    Δ2 = [1, 0, 0, 1, 0, 1]
+    S1 = Surv(T1, Δ1, :r)
+    S2 = Surv(T2, Δ2, :r)
+    M = merge(S1, S2)
+
+    stats1 = SurvivalAnalysis._padstats(S1.stats, unique_outcome_times(M))
+    @test isapprox(stats1.time, [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12])
+    @test isapprox(stats1.nrisk, [5, 4, 4, 3, 3, 2, 2, 1, 1, 0, 0])
```
Comment (RaphaelS1) [inline test/SurvivalEstimator.jl:19]: Not sure this works, at the first time point everyone (n=11) is at risk

File: src/SurvivalEstimator.jl [hunk 4]
```diff
@@ -338,6 +338,128 @@ function StatsBase.fit!(obj::KaplanMeier, Y::RCSurv)
     )
 end
 
+#-------------------
+# Log-rank tests
+#-------------------
+function update_wt(wt, nevents, nrisk, wtmethod)
```
Comment (RaphaelS1) [inline src/SurvivalEstimator.jl:344]: To be more consistent with design interface maybe instead `_update_wt` ?

File: src/SurvivalEstimator.jl [hunk 5]
```diff
@@ -338,6 +338,128 @@ function StatsBase.fit!(obj::KaplanMeier, Y::RCSurv)
...
+    if wtmethod == :logrank
+        return 1.0
+    elseif wtmethod == :wilcoxon
+        return nrisk
+    elseif wtmethod == :tw
+       return sqrt(nrisk)
+    elseif wtmethod == :peto
+        return wt * (1 - nevents / (nrisk + 1))
+    else
+        error("Unknown weight")
```
Comment (RaphaelS1) [inline src/SurvivalEstimator.jl:354]: Please make this more informative, what weights are allowed?

File: src/SurvivalEstimator.jl [hunk 6]
```diff
@@ -338,6 +338,128 @@ function StatsBase.fit!(obj::KaplanMeier, Y::RCSurv)
...
+    end
+
+    return u, V
+end
+
+"""
+    logrank_test(Y::RCSurv...; wtmethod=:logrank)
+
+Test the null hypothesis that two or more survival functions are identical.
+"""
```
Comment (RaphaelS1) [inline src/SurvivalEstimator.jl:398]: Please make this more informative, see other documentation. Include examples and make clear what the choices are for `wtmethod`

File: src/Surv.jl (hunk 2 above)
Comment (RaphaelS1) [reply src/Surv.jl:367]: Please also consider a more informative name so it's clear for developers what to use this for

**Review by RaphaelS1 — CHANGES_REQUESTED** [review]
Thanks a lot for this! Some comments about design choices and the `_padstats` method but happy to see all tests passed.

**Comment by RaphaelS1** [comment]
Please also get coverage to 100%

File: src/SurvivalEstimator.jl (hunk 1 above)
Comment (RaphaelS1) [reply src/SurvivalEstimator.jl]: @kshedden please merge this suggestion for better type consistency

**Comment by RaphaelS1** [comment]
Thanks @kshedden . New documentation makes it clearer what the `expandstats` function does, it was confusing looking at the example in `tests` but makes sense now. doctests are currently failing so please fix and then we can merge. Please also see @mschauer comment above. In future I think we should improve the output of the test for better consistency across the package but happy for now.

Closes https://github.com/RaphaelS1/SurvivalAnalysis.jl/issues/10

**Comment by RaphaelS1** [comment]
@all-contributors
please add @kshedden for doc, code.
please add @mschauer for review.

**Review by RaphaelS1 — APPROVED** [review]
Great , thanks!

**Pull request merged**